File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    if len(prices) < 2 {
        return 0
    }

    left, right := 0, 1

    maxProfit := 0

    for right < len(prices) {
        if prices[left] < prices[right] {
            profit := prices[right] - prices[left]
            maxProfit = max(maxProfit, profit)
        } else {
            left = right
        }
        right++
    }

    return maxProfit
}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.php
class Solution {

    /**
     * @param Integer[] $prices
     * @return Integer
     */
    function maxProfit($prices) {
        $len = count($prices);
        $left = 0;
        $max = 0;

        for ($right = 0; $right < $len; $right++) {
            if ($prices[$right] < $prices[$left]) {
                $left = $right;
            } else {
                $max = max($prices[$right] - $prices[$left], $max);
            }
        }

        return $max;
    }
}